src/utils/deepClone.ts

```diff
@@ -0,0 +1,25 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+const fromEntries = (it: Array<[string, any]>): any =>
+    it.reduce<any>(
+        (pv, [_key, _value]) =>
+            pv[_key] = typeof _value === "object" ? fromEntries(_value) : _value,
+        {});
+
+const ObjFromEntries = Object.fromEntries || fromEntries;
+
+export const deepClone = (o: unknown): any => {
+
+    const _clone = (_o: unknown): Array<[string, unknown]> =>
+        Object.entries(_o).map<[string, unknown]>(
+            ([_key, _value]) =>
+                typeof _value === "object" ? [_key, _clone(_value)] : [_key, _value]);
```

Comment:
if typeof _value is array, potentially the array items can be objects, so need recursing too?

Reply:
sorry, my bad, from entries does that.

Reply:
actually i rest my case. i need to run sone tests.

Reply:
I will add some tests

Reply:
I added a minimal repro test in the comment above.

Reply:
i have fixed `deepClone` and added some tests
feel free to check again